notifications_pipeline_steps/NotificationsPipelineDelivery.py

```diff
@@ -0,0 +1,12 @@
+from datetime import datetime
+import NotificationsPipelineStep
+class NotificationsPipelineDelivery(NotificationsPipelineStep.NotificationsPipelineStep):
+    @staticmethod
+    def process_message(message):
+        print message.current_step + " has received " + message.name + " " + str(message.id)
+        if datetime.utcnow() > message.expiration_time:
+            print message.name + " " + str(message.id) + " has expired. It will be dropped."
+        else:
+            log_file = open("logs/"+str(message.id)+".txt","w")
+            log_file.write(str(vars(message)))
+            log_file.close()
```

Comment:
Use python's logging module instead of a custom system.  Each message will get a single line in the log rather than its own file.  The existing version will very quickly use a ridiculous number of files, which can end up hitting system limits of one kind or another.